torch/csrc/jit/backends/nnapi/nnapi_backend_lib.cpp

```diff
@@ -17,23 +28,116 @@ class NnapiBackend : public PyTorchBackendInterface {
     return true;
   }
 
-  // Function stub
-  // TODO: implement compile
   c10::impl::GenericDict compile(
       c10::IValue processed,
       c10::impl::GenericDict method_compile_spec) override {
-    auto handles =
-        c10::Dict<std::string, std::vector<std::tuple<std::string, int64_t>>>();
+    auto dict = processed.toGenericDict();
+
+    // Prepare weights
+    auto weights = dict.at("weights").toTensorList();
+    for (int i = 0; i < weights.size(); i++) {
+      weights.set(i, weights.get(i).contiguous());
+    }
+    dict.insert("weights", weights);
+
+    // Save ser_model to member variable
+    ser_model_ = dict.at("ser_model").toTensor();
```

Comment:
Could you remind me why ser_model has to be a member variable?

Reply:
It was because ser_model is mutated in init(), and since init() is called once by execute(), the only way to carry those changes to later iterations is by making a member variable.
Although now that I'm looking at it again, execute() doesn't actually use ser_model. Removing the member variable might work too. I'll give it a try.

Reply:
yeah whatever we can remove from the state would be good.